lib/shipcloud/request/connection.rb

```diff
@@ -19,27 +19,28 @@ def setup_https
         @https.set_debug_output $stdout if Shipcloud.configuration.debug
       end
 
-      def request
-        https.start do |connection|
-          https.request(https_request)
+      def request(affiliate_id)
+        https.start do
+          https.request(https_request(affiliate_id))
         end
       end
 
       protected
 
-      def https_request
+      def https_request(affiliate_id)
+        affiliate = Shipcloud.api_headers.merge("Affiliate-ID" => affiliate_id)
```

Comment:
I'd suggest calling this `headers`. After the merge It's not just the affiliate ID anymore but the default headers.

```suggestion
        headers = Shipcloud.api_headers.merge("Affiliate-ID" => affiliate_id)
```